src/Workspaces/Core/Portable/PatternMatching/AllLowerCamelCaseMatcher.cs

```diff
@@ -240,9 +262,10 @@ private bool IsBetter(CamelCaseResult result, CamelCaseResult? currentBestResult
             private bool LowercaseSubstringsMatch(
                 string s1, int start1, string s2, int start2, int length)
             {
+                var textInfo = _textInfo;
                 for (var i = 0; i < length; i++)
                 {
-                    if (char.ToLower(s1[start1 + i]) != char.ToLower(s2[start2 + i]))
```

Comment:
calling char.ToLower was expensively having to call Thread.CurrentThread.CurrentCulture.  When done millions of times, that added up to a lot of overhead.